src/Package/Impl/Sql/Publish/SProcGenerator.cs

```diff
@@ -0,0 +1,186 @@
+                sw.WriteLine("(");
+                sw.WriteLine(Invariant($"[{SProcColumnName}] NVARCHAR(64),"));
+                sw.WriteLine(Invariant($"[{RCodeColumnName}] NVARCHAR(max)"));
+                sw.WriteLine(")");
+            }
+            targetProjectItem.ProjectItems.AddFromFile(creatTableScriptFile);
+        }
+
+        /// <summary>
+        /// Generates SQL post deployment script that pushes R code into a table
+        /// as well as 
+        /// </summary>
+        private void CreatePostDeploymentScript(SqlSProcPublishSettings settings, EnvDTE.Project targetProject, string targetFolder, EnvDTE.ProjectItem targetProjectItem) {
+            var postDeploymentScript = Path.Combine(targetFolder, PostDeploymentScriptName);
+
+            using (var sw = new StreamWriter(postDeploymentScript)) {
+                sw.WriteLine(Invariant($"INSERT INTO {settings.TableName}"));
+
+                for (int i = 0; i < settings.Files.Count; i++) {
+                    var filePath = settings.Files[i];
+
+                    var sprocName = settings.SProcNames[filePath];
+                    if (!string.IsNullOrEmpty(sprocName)) {
+                        var content = GetRFileContent(filePath);
+                        sw.Write(Invariant($"VALUES ('{sprocName}', '{content}')"));
+                        if (i < settings.Files.Count - 1) {
+                            sw.Write(',');
+                        }
+                        sw.WriteLine(string.Empty);
+                    }
+                }
+            }
+            var item = targetProjectItem.ProjectItems.AddFromFile(postDeploymentScript);
+            item.Properties.Item("BuildAction").Value = "PostDeploy";
+        }
+
+        /// <summary>
+        /// Replaces procedure name, R Code and the SQL query placeholders with actual values
+        /// </summary>
+        private string FillSprocInlineTemplate(string rFilePath, string sprocName) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var rCode = GetRFileContent(rFilePath);
+            rCode = rCode.EndsWithOrdinal(Environment.NewLine) ? rCode : rCode + Environment.NewLine;
+            sprocTemplate = sprocTemplate.Replace(RCodeTemplate, Environment.NewLine + rCode);
+
+            var sqlQuery = GetSqlFileContent(rFilePath.ToQueryFilePath()).Trim();
+            return sprocTemplate.Replace(InputQueryTemplate, sqlQuery);
+        }
+
+        private string FillSprocTableTemplage(string rFilePath, string sprocName, string codeTableName) {
+            var sprocTemplateFile = rFilePath.ToSProcFilePath();
+            var sprocTemplate = GetSqlFileContent(sprocTemplateFile);
+
+            var format =
+@"BEGIN
+DECLARE @RCodeQuery NVARCHAR(max);
+DECLARE @RCode NVARCHAR(max);
+DECLARE @ParmDefinition NVARCHAR(max);
+
+SET @RCodeQuery = 'SELECT @RCodeOUT = RCode FROM {0} WHERE SProcName = ''{1}''';
```

Comment:
And you might want `N` prefix on the string literal, in case any Unicode chars sneak in.